Lib/test/test_list.py

```diff
@@ -256,6 +256,19 @@ def __eq__(self, other):
         lst = [X(), X()]
         X() in lst
 
+    def test_constant_list_subscript(self):
+        # testing constant list subscripting
+        code = compile("[5, 7, 10, 11][2]", "<string>", "eval")
+        self.assertEqual(code.co_consts, (10,))
+        self.assertInBytecode(code, "LOAD_CONST", 0)
+        self.assertNotInBytecode(code, "BUILD_LIST")
+        self.assertNotInBytecode(code, "BINARY_SUBSCR")
+
+        code = compile("[][0]", "<string>", "eval")
+        self.assertEqual(code.co_consts, (0,))
+        self.assertInBytecode(code, "BUILD_LIST")
+        self.assertInBytecode(code, "BINARY_SUBSCR")
```

Comment:
This test has a problem that many other tests in this file have - the NotInBytecode check is flaky because we sometimes rename a bytecodes and this is just a string.

A way to make this test less flaky is to add the case where the index is not const, and check that "BUILD_LIST" and "BINARY_SUBSCR" are in the bytecode.  Then this part of the test will fail if the bytecode is renamed.

Reply:
Is everything good now?

Reply:
The test looks fine.

It remains to resolve the questions of whether this optimization should go in at all (i.e., does this code show up in real programs?)